lab3.c

```diff
@@ -0,0 +1,141 @@
+#include "lab3.h"
+
+void freeInfo(Info *info) {
+    free(info->from);
+    free(info->where);
+    free(info);
+}
+
+void *copyingFile(void *arg) {
+    Info *info = (Info *)arg;
+
+    char buffer[BUFFER_SIZE];
+    ssize_t countRead, countWrite;
+
+    errno = 0;
+    int fileFrom = open(info->from, O_RDONLY);
+    
+    while (errno == EMFILE) {
```

Comment:
bad approach to check errno without check of return value. It might lead to many strange problems. Please fix it and explain what is the problem I mentioned.

Reply:
Errno is set by functions in case of an error. So before checking errno, we should make sure that the function returned an error. Otherwise, the program will have undefined behavior, we may miss an error, or the errno value will not match the expected one

Reply:
yep